### PR TITLE
FORNO-1745: Remove use of `esc_html()` when handling mail failures

### PR DESCRIPTION
**Pull request by t-wright**
## Description

In https://github.com/Automattic/vip-go-mu-plugins/pull/5182 we added handling for failures when sending mail from unmapped domains. In this PR we're removing the use of `esc_html` on the email body and other fields which was added in that PR since the use of this function could lead to undesirable escaping in this context.

### CODE REVIEW

**Review by sanmai — APPROVED** [review]
Reviewed the code, looks sane.

File: vip-mail.php [hunk 1]
```diff
@@ -142,28 +142,34 @@ public function filter_wp_mail_from() {
...
-					$to          = is_array( $error_data['to'] ) ? array_map( 'esc_html', $error_data['to'] ) : esc_html( $error_data['to'] );
-					$subject     = esc_html( $error_data['subject'] );
-					$message     = esc_html( $error_data['message'] );
-					$headers     = is_array( $error_data['headers'] ) ? array_map( 'esc_html', $error_data['headers'] ) : esc_html( $error_data['headers'] );
-					$attachments = is_array( $error_data['attachments'] ) ? array_map( 'esc_url', $error_data['attachments'] ) : esc_url( $error_data['attachments'] );
+				$to          = $error_data['to'] ?? null;
+				$subject     = $error_data['subject'] ?? null;
+				$message     = $error_data['message'] ?? null;
+				$headers     = $error_data['headers'] ?? null;
+				$attachments = $error_data['attachments'] ?? null;
```
Comment (saroshaga) [inline vip-mail.php:159]: We're removing `esc_url` from here

Comment (t-wright) [reply vip-mail.php:159]: Yes, I think that one should have been okay, but probably unnecessary.

Comment (saroshaga) [reply vip-mail.php:159]: got it! thanks!

**Review by saroshaga — APPROVED** [review]

**Pull request merged**